### PR TITLE
feat: 🎸 Load spinner when selecting gcloud store. Add key on td

### LINKED ISSUE #1029: Study List data formatting
@pieper has a DICOM store on Google Cloud which fails every time it gets to Page 2 of the study list. It looks like it's because React is trying to render an Object inside one of the table cells in the study list. We need to ensure that each of the fields in the study object used below are renderable as Strings by React.

https://github.com/OHIF/Viewers/blob/659649e10b3732d4ce9a0d851390bb75786727eb/platform/ui/src/components/studyList/StudyList.js#L224-L253

> Uncaught Invariant Violation: Objects are not valid as a React child (found: object with keys {}). If you meant to render a collection of children, use an array instead.
    in td (created by StudyList)
    in tr (created by StudyList)
    in tbody (created by StudyList)
    in table (created by StudyList)
    in div (created by StudyList)
    in div (created by StudyList)
    in StudyList (created by TranslatedComponent)
    in TranslatedComponent (created by withI18nextTranslation(TranslatedComponent))
    in withI18nextTranslation(TranslatedComponent) (created by WrapperComponent)
    in I18nextProvider (created by WrapperComponent)

### PR DESCRIPTION
**Pull request by ladeirarodolfo**
### Includes
- Added loader spinner into study list when searching/filtering list
- Added React keys to table td tags
- Robust implementation preventing errors when cell value is not a string(it fixes the issue 1029)
- Reduced amount of component render
- Minor improvement on Cypress implementation

It closes #1029

### CODE REVIEW

File: platform/ui/src/components/studyList/StudyList.js [hunk 1]
```diff
@@ -16,6 +16,69 @@ import { withTranslation } from '../../utils/LanguageProvider';
...
+
+function getNoListFragment(studies, error, loading) {
+  if (loading) {
+    return (
+      <div className="loading">
+        <StudyListLoadingText />
+      </div>
+    );
+  } else if (error) {
+    return <div className="notFound">There was an error fetching studies</div>;
```
Comment (swederik) [inline platform/ui/src/components/studyList/StudyList.js]: Can we wrap these in translation strings (this.props.t()) so they can be internationalized?

File: platform/ui/src/components/studyList/StudyList.js [hunk 2]
```diff
@@ -399,25 +441,23 @@ class StudyList extends Component {
               </tr>
             </thead>
             <tbody id="studyListData">
-              {this.props.studies.map(study => {
-                return this.renderTableRow(study);
-              })}
+              {!noListFragment
+                ? this.props.studies.map((study, index) => {
```
Comment (swederik) [inline platform/ui/src/components/studyList/StudyList.js]: I think you can just write `this.props.studies.map(this.renderTableRow)`

**Comment by dannyrb** [comment]
Netlify:

image

I have the branch checked out locally -- taking a peek.

**Comment by ladeirarodolfo** [comment]
I have the fix for CI failing.. I will push soon, after some technical discussion with QA team

File: platform/ui/src/components/studyList/StudyList.js [hunk 3]
```diff
@@ -110,25 +177,27 @@ class StudyList extends Component {
 
   getBlurHandler(key) {
     return event => {
+      debugger;
```
Comment (swederik) [inline platform/ui/src/components/studyList/StudyList.js]: Forgot a debugger

File: platform/ui/src/components/studyList/StudyList.js [hunk 4]
```diff
@@ -177,17 +177,38 @@ class StudyList extends Component {
...
     return event => {
-      debugger;
       this.delayedSearch.cancel();
       this.setSearchData(key, event.target.value);
     };
   }
 
+  // Should update state method.
+  // Used to prevent relying on shouldComponentUpdate React method or Pure component implementations of shallow comparison.
+  shouldUpdateStateValue(stateValue, stateValueToCompare) {
```
Comment (swederik) [inline platform/ui/src/components/studyList/StudyList.js]: Can / should we use something like lodash isEqual instead?

**Comment by dannyrb** [comment]
CI is passing

**Pull request merged**